test/common/formatter/substitution_formatter_test.cc

```diff
@@ -3109,8 +3109,7 @@ TEST(SubstitutionFormatterTest, ParserFailures) {
       "RESP(FIRST)%",
       "%REQ(valid)% %NOT_VALID%",
       "%REQ(FIRST?SECOND%",
-      "%%",
-      "%%HOSTNAME%PROTOCOL%",
+      "%HOSTNAME%PROTOCOL%",
```

Comment:
Do we have a test case that verifies that `%HOSTNAME%%PROTOCOL%` parses correctly. From the code it looks like it would, but it would be good to add a test case like this if it does not exist.

Reply:
I've added one in commit [9c1f769](https://github.com/envoyproxy/envoy/pull/20093/commits/9c1f769312c35a6abbc9bd3585bc2ace746d29b9)